### PR TITLE
Add support for NetCDF/Time-partitioned CSV QLAT inputs

### PR DESCRIPTION
**Pull request by program--**
This PR adds support for reading NetCDF QLAT outputs from `ngen` via `xarray`.

The NetCDF schema for the new outputs are as follows:
```
netcdf 201512010000NEXOUT {
dimensions:
        feature_id = 3 ;
variables:
        string feature_id(feature_id) ;
                feature_id:description = "Contributing Nexus ID" ;
        double q_lateral(feature_id) ;
                q_lateral:description = "Runoff into channel reach" ;
                q_lateral:units = "m3 s-1" ;
        string time ;
                time:description = "Output timestamp for this contribution" ;
                time:standard_name = "time" ;
                time:format = "%Y%m%s%H%M" ;
data:

 feature_id = "nex-285284", "nex-285285", "nex-285286";

 q_lateral = 8.53977600158463, 7.45141160538848, 5.80901770862798;

 time = "201512010000";
}
```
> The corresponding CSV header is `feature_id, <time>` (i.e. `feature_id, 201512010000`).

> Related: https://github.com/NOAA-OWP/ngen/issues/605, https://github.com/NOAA-OWP/ngen/pull/612

## Additions

- Add NetCDF reading to both `AbstractNetwork.read_file` and `HYFeaturesNetwork.read_file`.
- Adds timestamp parsing from file names for `*NEXOUT` glob filters.

## Changes

- Modifies `build_qlateral_array` when reading a q_lateral file to remove `nex-` from the prefix.
  > This probably should follow the logic of trying to split on `-`, and remove everything prefixing it...

## Testing

- Adds a unit test for reading from `ngen`-outputted Q Lateral files (both CSV and NetCDF), and ensures forcing sets can be assembled.

## Todos

- [X] ~Add logic for CSV heuristics~ Unneeded if relying on glob filter
- [X] Add unit tests

## Checklist

- [X] PR has an informative and human-readable title
- [X] Changes are limited to a single goal (no scope creep)
- [ ] Code can be automatically merged (no conflicts)
- [X] Code follows project standards (link if applicable)
- [x] Passes all existing automated tests
- [x] Any _change_ in functionality is tested
- [x] New functions are documented (with a description, list of inputs, and expected output)
- [x] Placeholder code is flagged / future todos are captured in comments
- [x] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [x] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)
- [x] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### CODE REVIEW

**Comment by program--** [comment]
@shorvath-noaa if you get a chance, can you pull this PR and ensure the tests and in-memory df schema are as expected -- and maybe do a quick review in case there are things that I missed for this integration?

File: src/troute-network/troute/HYFeaturesNetwork.py [hunk 1]
```diff
@@ -511,16 +512,19 @@ def build_qlateral_array(self, run,):
...
+            nexuses_lateralflows_df = pd.concat(dfs, axis=1) 
             
             # Take flowpath ids entering NEXUS and replace NEXUS ids by the upstream flowpath ids
-            qlats_df = pd.concat( (nexuses_lateralflows_df.loc[int(k)].rename(v)
-                                for k,v in self.downstream_flowpath_dict.items() ), axis=1
-                                ).T 
+            qlats_df = pd.concat(
+                (nexuses_lateralflows_df.loc[int(k)].rename(v) for k,v in self.downstream_flowpath_dict.items() if k in nexuses_lateralflows_df.index.to_list()),
+                axis=1
+            ).T
```
Comment (shorvath-noaa) [inline src/troute-network/troute/HYFeaturesNetwork.py:527]: this seems to be entering an infinite loop when I run t-route with *CHRTOUT files as forcing. If you change this to:
`qlats_df = nexuses_lateralflows_df.rename(index=self.downstream_flowpath_dict)`
it works with *CHRTOUT files and still passes your pytest file.

Comment (program--) [reply src/troute-network/troute/HYFeaturesNetwork.py]: Awesome, I think this works well. Implemented in ebfdd6d.

**Review by shorvath-noaa — APPROVED** [review]

**Comment by program--** [comment]
@shorvath-noaa any update on when this can be merged? NOAA-OWP/ngen#612 is soft-blocked on this (but no immediate rush). Thanks!!

**Pull request merged**